v3/src/components/graph/models/graph-content-model.ts

```diff
@@ -139,6 +154,40 @@ export const GraphContentModel = DataDisplayContentModel
         xScale,
         yScale
       }
+    },
+    binDetails(_binAlignment?: number, _binWidth?: number) {
```

Comment:
@kswenson I ended up refactoring `binDetails` following your (off-Github) suggestion of adding an option for specifying if the width and alignment should be initialized from the data. I did still split out the computation of bin width since that isn't always needed and moving the code for it helped simplify binDetails a little. What do you think of that?